xskillscore/core/probabilistic.py

```diff
@@ -820,3 +822,239 @@ def _reliability(o, f, bin_edges):
 
     # Move samples to a coordinate
     return _add_as_coord(rel, samp, coordinate_suffix="samples")
+
+
+def roc(
+    observations,
+    forecasts,
+    bin_edges="continuous",
+    dim=None,
+    drop_intermediate=False,
+    return_results="area",
+):
+    """Computes the relative operating characteristic for a range of thresholds.
+
+    Parameters
+    ----------
+    observations : xarray.Dataset or xarray.DataArray
+        Labeled array(s) over which to apply the function.
+        If ``bin_edges=='continuous'``, observations are binary.
+    forecasts : xarray.Dataset or xarray.DataArray
+        Labeled array(s) over which to apply the function.
+        If ``bin_edges=='continuous'``, forecasts are probabilities.
+    bin_edges : array_like, str, default='continuous'
+        Bin edges for categorising observations and forecasts.
+        Bins include the left most edge, but not the right. ``bin_edges`` will be
+        sorted in ascending order. If ``bin_edges=='continuous'``, calculate
+        ``bin_edges`` from forecasts, equal to
+        ``sklearn.metrics.roc_curve(f_boolean, o_prob)``.
+    dim : str, list
+        The dimension(s) over which to compute the contingency table
+    drop_intermediate : bool, default=True
+        Whether to drop some suboptimal thresholds which would not appear on a plotted
+        ROC curve. This is useful in order to create lighter ROC curves.
+    return_results: str, default='area'
+        Specify how return is structed:
+
+            * 'area': return only the ``area under curve`` of ROC
+            * 'all_as_tuple': return ``true positive rate`` and ``false positive rate`` at each
+                bin and area under the curve of ROC as tuple
+            * 'all_as_metric_dim': return ``true positive rate`` and ``false positive rate`` at
+               each bin and ``area under curve`` of ROC concatinated into new ``metric``
+               dimension
+
+    Returns
+    -------
+    xarray.Dataset or xarray.DataArray : reduced by dimensions ``dim``. See
+    ``return_results`` parameter. ``true positive rate`` and ``false positive rate`` contain ``probability_bin`` dimension with ascending ``bin_edges`` as coordinates.
```

Comment:
```suggestion
    xarray.Dataset or xarray.DataArray : reduced by dimensions ``dim``.
    See ``return_results`` parameter.
    ``true positive rate`` and ``false positive rate`` contain
    ``probability_bin`` dimension with ascending ``bin_edges`` as coordinates.
```

Docs build a bit funny for this. May have to test.